job/dhcp.go

```diff
@@ -139,15 +139,15 @@ func (j Job) setPXEFilename(rep *dhcp4.Packet, isPacket, isARM, isUEFI, isHTTPCl
 	}
 
 	var filename string
-	if !isPacket {
+	if !isTinkerbellIPXE {
 		if j.PArch() == "hua" || j.PArch() == "2a2" {
-			filename = "ipxe/snp-hua.efi"
+			filename = "snp.efi"
 		} else if isARM {
-			filename = "ipxe/snp-nolacp.efi"
+			filename = "snp.efi"
```

Comment:
This caught my eye. Have you been able to test this against the hua/2a2 machines in our fleet? The snp-hua.efi was definitely a hard requirement as no other ipxe build would boot. If this is no longer the case then we can drop the `PAarch` check and should be called out in the commit message explicitly.

Reply:
I have not tested against any hua/2a2. Any idea how to build this iPXE binary? it should/needs to be built in the open like the others in github.com/tinkerbell/ipxedust but seems it's just been saved in [git-lfs](https://github.com/tinkerbell/boots/blob/ba3a3fef424ebfd7125b08ae99dcb9631bc911a8/.gitattributes).

Reply:
I haven't tried building this recently (it's been at least 2 years) but every time I tried it failed to boot. As far as I know its a magic golden image, one of a kind. That being said it *would* be good to see if recent ipxe builds will boot correctly on them.

Reply:
Hey @mmlb, I tested this against Equinix Metal machines with plan type c1.large.arm.xda ([boots ref](https://github.com/tinkerbell/boots/blob/aefc60d0f405bd0fc0c56fc000f7b302801171ff/job/helpers.go#L47)) and against plan type Type HUA (baremetal_hua) ([boots ref](https://github.com/tinkerbell/boots/blob/aefc60d0f405bd0fc0c56fc000f7b302801171ff/job/helpers.go#L53)) and they were both successful.

Reply:
*awesome*!